src/ristretto/ristretto_keys.rs

```diff
@@ -74,29 +74,51 @@ impl borsh::BorshDeserialize for RistrettoSecretKey {
 //-----------------------------------------   Ristretto Secret Key    ------------------------------------------------//
 impl SecretKey for RistrettoSecretKey {
     const KEY_LEN: usize = 32;
+    const WIDE_REDUCTION_LEN: usize = 64;
 
     /// Return a random secret key on the `ristretto255` curve using the supplied CSPRNG.
     fn random<R: Rng + CryptoRng>(rng: &mut R) -> Self {
         RistrettoSecretKey(Scalar::random(rng))
     }
+
+    /// Return a secret key computed by wide reduction on a byte array
+    /// If the byte array is not exactly 64 bytes, returns an error
+    fn from_bytes_wide(bytes: &[u8]) -> Result<Self, ByteArrayError> {
+        if bytes.len() != Self::WIDE_REDUCTION_LEN {
+            return Err(ByteArrayError::IncorrectLength {});
+        }
+
+        // We need a slice of the correct length, which can't be enforced by the function signature
+        let mut bytes_copied = Zeroizing::new([0u8; Self::WIDE_REDUCTION_LEN]);
+        bytes_copied.copy_from_slice(bytes);
+
+        Ok(RistrettoSecretKey(Scalar::from_bytes_mod_order_wide(&bytes_copied)))
+    }
 }
 
 //-------------------------------------  Ristretto Secret Key ByteArray  ---------------------------------------------//
 
 impl ByteArray for RistrettoSecretKey {
-    /// Create a secret key on the Ristretto255 curve using the given little-endian byte array. If the byte array is
-    /// not exactly 32 bytes long, `from_bytes` returns an error. This function is guaranteed to return a valid key
-    /// in the group since it performs a mod _l_ on the input.
+    /// Return a secret key computed from a canonical byte array
+    /// If the byte array is not exactly 32 bytes, returns an error
+    /// If the byte array does not represent a canonical encoding, returns an error
     fn from_bytes(bytes: &[u8]) -> Result<RistrettoSecretKey, ByteArrayError>
     where Self: Sized {
-        if bytes.len() != 32 {
+        if bytes.len() != Self::KEY_LEN {
             return Err(ByteArrayError::IncorrectLength {});
         }
-        let mut a = [0u8; 32];
-        a.copy_from_slice(bytes);
-        let k = Scalar::from_bytes_mod_order(a);
-        a.zeroize();
-        Ok(RistrettoSecretKey(k))
+
+        // We need a slice of the correct length, which can't be enforced by the function signature
```

Comment:
```suggestion
        // We need an array of the correct length, which can't be enforced by the function signature
```

.. and anyway `Scalar::from_canonical_bytes` takes ownership

Reply:
Yep. I copied the other comment, which was inaccurate in this case. Removed.